custom_components/ha-glocaltokens/api.py

```diff
@@ -52,43 +52,46 @@ def get_android_id(self):
     def create_url(self, ip, port, api_endpoint):
         url = 'https://{ip}:{port}/{endpoint}'.format(
             ip=ip,
-            port=port,
+            port=str(port),
             endpoint=api_endpoint
         )
         return url
 
-    def get_alarms_and_timers_from(self, ip, port, token, endpoint):
-        url = self.create_url(ip, port, endpoint)
-        HEADERS[HEADER_CAST_LOCAL_AUTH] = token
+    def get_alarms_and_timers_from(self, device, endpoint):
+        url = self.create_url(device[DEVICE_IP], device[DEVICE_PORT], endpoint)
+        _LOGGER.debug("For device {device} - {url}".format(device=device[DEVICE_NAME], url=url))
+        HEADERS[HEADER_CAST_LOCAL_AUTH] = device[TOKEN]
         response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT) # verify=False is need to avoid SSL security checks. Othervise it will fail to connect
 
-        return response
+        if response.status_code != HTTP_OK:
+            _LOGGER.error("For device {device} - API returned {error}".format(device=device[DEVICE_NAME], error=response.status_code))
+            return False
+        else:
+            return response.json()
 
     def get_google_devices_information(self):
-
+        _LOGGER.debug("Fetching data...")
         devices = self._client.get_google_devices_json()
 
         for device in devices:
             # To avoid keyerror's
-            device[TIMERS] = []
-            device[ALARMS] = []
-
-            ip = '192.168.0.205'
-            port = str(PORT)
-            token = device[TOKEN]
-            response = self.get_alarms_and_timers_from(ip, port, token, API_ENDPOINT_ALARMS) # IP
-
-            if response.status_code != HTTP_OK:
-                _LOGGER.error("For device {device} - API returned {error}".format(device=device[DEVICE_NAME], error=response.status_code))
-                continue
-
-            result = response.json()
-
-            if TIMERS not in result and ALARMS not in result:
-                _LOGGER.error("For device {device} - {error}".format(device=device[DEVICE_NAME], error=API_RETURNED_UNKNOWN))
-                continue
-
-            device.update(result)
+            timers = []
+            alarms = []
+
+            device[DEVICE_IP] = '192.168.0.205' # For testing purpose only
+            device[DEVICE_PORT] = PORT # For testing purpose only
+            result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
+            if result != False:
```

Comment:
Then this could just look like: `if result:`
Very pythonic 😄

Reply:
Python amazes me every time by how simple and easy it is, i will resolves both of these now